autoversioner/cmd/tagger/tag.go

```diff
@@ -42,38 +42,38 @@ func main() {
 // create ref /refs/tags/<tagname>; createtag just creates a tag object
 // refer https://developer.github.com/v3/git/tags/#create-a-tag-object
 func AddTag(ctx context.Context) (*github.Reference, error) {
-	tagger, err := tagger.NewClient(ctx, *tokenFile)
+	tagger, err := ghub.NewClient(ctx, *tokenFile)
 	if err != nil {
-		return nil, fmt.Errorf("Error creating tagger: %+v\n", err)
+		return nil, fmt.Errorf("error creating tagger: %+v", err)
 	}
 	tag, err := tagger.CreateTag(ctx, *org, *repo, *tag, *sha, *message, *botUser, *botEmail)
 	if err != nil {
-		return nil, fmt.Errorf("Error while tagging: %+v\n", err)
+		return nil, fmt.Errorf("error while tagging: %+v", err)
 	}
 	fmt.Printf("Added tag correctly: %+v\n", tag)
 	ref, err := tagger.CreateRef(ctx, *org, *repo, fmt.Sprintf("refs/tags/%s", *tag.Tag), *tag.SHA)
 	if err != nil {
-		return nil, fmt.Errorf("Error while creating ref: %+v", err)
+		return nil, fmt.Errorf("error while creating ref: %+v", err)
 	}
 
 	return ref, nil
 }
 
 func validateFlags() error {
```

Comment:
doesn't need to be a function, can be a single if with || conditions

Reply:
initially i had them in || conditions but for readability sake i just moved it to a separate function. also introducing this function would help us to do extra validations; if need be

Reply:
> > strategy is overly complicated bc it supports arbitrary dates - we will only ever be building for todays date and building the newest possible version, so i think it can be simplified.
> 
> it was not so much to support arbitary date. It was one of those corner cases for commits being merged just before midnight and there is delayed time interval between when the commit was merged and when the prowjob was kicked in(may be due to network); and this delay caused
> dates to change. In this case; the commit date and it version tag will be inconsistent.

we should only generate version once and hand to the build workflow and use same for post-build-tagging. that will either be before midnight or it will be after - we should not care when it was committed, only when we are running the build.

Reply:
idk why this got attached to this comment, sorry

Reply:
i think there is misunderstanding; we are generating the version only once(and actually, with the current algorithm, even if we generate the version for a particular commit multiple times, it will generate the same version); i introduced this so that the commit date and the version date are consistent. 

without this, how would you find the build number for a particular commit if, for any reason, there is a build kicked off on next day(or any other day for that matter)?

Reply:
you will always get 'today' component of the version, and only ever need to determine what build number for today we're on - you can look at latest git tag for this, regardless of time or date. if the latest git tag isn't today, then the build number is 00.

Reply:
iiuc, dont we end up in the same synchronization problem that we talked yesterday. 

2 parallel builds are running; one for a commit before midnight and one for a commit after midnight. we take the today component of the version and for the build number we check the tag; we do not get any tag for today; so we assign 00 to both builds.

also, i want to understand what part of the algorithm do you think is complicated?